core/components/handler/handler.go

```diff
@@ -633,6 +634,9 @@ func (h component) consumeDown(appEUI []byte, devEUI []byte, dataRate string, bu
 			stats.MarkMeter("handler.downlink.pull")
 			downType := lorawan.UnconfirmedDataDown
 			ack := (upType == lorawan.ConfirmedDataUp)
+			if bundle.Packet.(*core.DataUpHandlerReq).FCntUpReset {
+				bundle.Entry.FCntDown = 0
```

Comment:
After it has been used in h.buildDownlink to set downlink.Payload.MACPayload.FHDR.FCnt.